src/battle_util.c

```diff
@@ -8353,6 +8353,37 @@ void UndoFormChange(u32 monId, u32 side)
     }
 }
 
+// Same as UndoFormChange but doesn't revert Mimikyu
+void UndoFormChangeOnSwitch(u32 monId, u32 side)
```

Comment:
This should be done in a single function I think. It would be better to modify the UndoFormChange function, so it takes an additional argument which notifies whether to include Mimikyu or not.
Then, the species table would be modified with Mimikyu as the first entry, and depending on the argument the iteration through the array would happen either from the first entry or the second(ignoring Mimikyu.)

Reply:
Oh that's much better than a separate function, will do that.